pkg/kubectl/cmd/drain.go

```diff
@@ -0,0 +1,156 @@
+		Short:   "Drain nodes in preparation for maintenance",
+		Long:    drain_long,
+		Example: drain_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(
+				RunDrain(f, out, cmd, args, options))
+		},
+	}
+	cmd.Flags().BoolVar(&options.Force, "force", false, "Delete unreplicated pods.")
+	cmd.Flags().IntVar(&options.GracePeriodSeconds, "grace-period", -1, "Period of time in seconds given to each pod to terminate gracefully. Ignored if negative.")
+	return cmd
+}
+
+func RunDrain(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *DrainOptions) error {
+	if len(args) != 1 {
+		return cmdutil.UsageError(cmd, fmt.Sprintf("USAGE: %s [flags]", cmd.Use))
+	}
+	options.NodeName = args[0]
+
+	c, err := f.Client()
+	if err != nil {
+		return err
+	}
+	if err = maybeLameNode(c, out, options); err != nil {
+		return err
+	}
+	pods, err := c.Pods(api.NamespaceAll).List(nil, fields.Set{"spec.nodeName": options.NodeName}.AsSelector())
+	if err != nil {
+		return err
+	}
+	if err = scanForReplicationControllers(c, out, options, pods); err != nil {
+		return err
+	}
+	if err = deletePods(c, out, options, pods); err != nil {
+		return err
+	}
+	return nil
+}
+
+func maybeLameNode(c *client.Client, out io.Writer, options *DrainOptions) error {
+	node, err := c.Nodes().Get(options.NodeName)
+	if err != nil {
+		return err
+	}
+	if node.Spec.Unschedulable {
+		fmt.Fprintf(out, "%s already unschedulable\n", node.Name)
+	} else {
+		node.Spec.Unschedulable = true
+		newNode, err := c.Nodes().Update(node)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(out, "%s marked unschedulable\n", newNode.Name)
+	}
+	return nil
+}
+
+func scanForReplicationControllers(c *client.Client, out io.Writer, options *DrainOptions, pods *api.PodList) error {
+	rcs, err := c.ReplicationControllers(api.NamespaceAll).List(nil, nil)
+	if err != nil {
+		return err
+	}
+	unreplicatedPodNames := []string{}
+
+	for _, pod := range pods.Items {
+		replicated := false
+		for _, rc := range rcs.Items {
+			selector := labels.SelectorFromSet(rc.Spec.Selector)
+			if selector.Matches(labels.Set(pod.Labels)) {
+				replicated = true
+				break
+			}
+		}
+		if !replicated {
+			unreplicatedPodNames = append(unreplicatedPodNames, pod.Name)
+		}
+	}
+	if len(unreplicatedPodNames) > 0 {
+		joined := strings.Join(unreplicatedPodNames, ", ")
+		if options.Force {
+			fmt.Fprintf(out, "WARNING: Deleting these unreplicated pods: %s\n", joined)
+		} else {
+			return fmt.Errorf("Refusing to continue due to unreplicated pods: %s", joined)
+		}
+	}
+	return nil
+}
+
+func deletePods(c *client.Client, out io.Writer, options *DrainOptions, pods *api.PodList) error {
+	gracePeriodSeconds := int64(options.GracePeriodSeconds)
+	deleteOptions := api.DeleteOptions{GracePeriodSeconds: &gracePeriodSeconds}
+
+	for _, pod := range pods.Items {
+		fmt.Fprintf(out, "DELETE %v\n", pod.Name)
+		err := c.Pods(pod.Namespace).Delete(pod.Name, &deleteOptions)
```

Comment:
Not sure what you mean by "back", but it will delete the pods and won't touch the RCs.